client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js

```diff
@@ -11,81 +11,130 @@ import {
-        );
-
-        return {
-          label: province.text,
-          data: formatted_yearly_tp,
-        };
-      }
-    })
-    .filter((data) => !_.isUndefined(data))
-    .value();
 
-export const declare_tp_by_region_panel = () => declare_panel({
-  panel_key: "tp_by_region",
-  levels: ["gov", "dept"],
-  panel_config_func: (level, panel_key) => ({
-    depends_on: ['orgTransferPaymentsRegion'],
-    calculate: function(subject, info, options){
-      const { orgTransferPaymentsRegion } = this.tables;
-      return calculate_common(subject, orgTransferPaymentsRegion);
-    },
-    
-    render(render_args){
-      const {
-        calculations,
-        footnotes,
-        sources,
-      } = render_args;
-      const { panel_args } = calculations;
-      const { data } = panel_args;
+class TPMap extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      show_per_capita: false,
+      loading: true,
+      population: {},
+    };
+  }
+  componentDidMount(){
+    loadPopulation()
+      .then( 
+        (transformed) => this.setState({
+          loading: false,
+          population: transformed,
+        })
+      );
+  }
+  render(){
+    const { calculations, footnotes, sources } = this.props;
+    const { show_per_capita, loading, population } = this.state;
+
+    if (loading) {
+      return (
+        <div style = {{position: "relative", height: "80px", marginBottom: "-10px"}}>
+          <SpinnerWrapper/>
+        </div>);
+    } else {
+      const { tables } = calculations.panel_args;
+      const { tp } = tables;
+
+      const changeState = () => {
+        this.setState({ show_per_capita: !show_per_capita });
+      };
+
+      const get_subject_data_for_year = (year) => tp.prov_code(
+        year,
+        calculations.subject.level === 'dept' && calculations.subject.id
+      );
+
+      //generating both data sets, as a11y table will need both later anyways
+      const tp_data = std_years.map(get_subject_data_for_year);
+      const tp_pc_data = std_years.map((year, i) => {
+        const single_year_tp_data = get_subject_data_for_year(year);
+        const result = _.chain(_.keys(single_year_tp_data))
+          .pullAll(["na", "abroad"])
+          .map((prov) => {
+            return [prov, single_year_tp_data[prov]/population[prov][i]];
+          })
+          .fromPairs()
+          .value();
+        return result;
+      });
+      
+      const data = (show_per_capita) ? tp_pc_data : tp_data;
+
       const current_year_data = _.last(data);
+      
+      //organize data for colour scale
+      const max = _.chain(data)
+        .last()
+        .values()
+        .max()
+        .value();
+      const color_scale = d3.scaleLinear()
+        .domain([0, max])
+        .range([0.2, 1]);
 
       const largest_prov = _.chain(current_year_data)
         .keys()
-        .maxBy( (prov) => current_year_data[prov] )
+        .maxBy((prov) => current_year_data[prov])
```

Comment:
The extra spaces here are something I tend to do, we'll be stomping on each other's toes if we're switching the code style back and forth, haha.

I try to balance out spaces within nested parentheses/brackets/etc. to keep space between, at a minimum, brackets of the same type. Makes it easier to scan for pairs and avoid miss-counting them while making edits. 

On a related note, I take it as a rule of thumb that if you ever have a line that ends in something like `) ) )` then it's a case where you should wrap and indent some of those levels of nesting, even if the total line isn't particularly long.

Reply:
Added clear spacing
Done